### ISSUE TITLE
Questionable CBMs which maybe should be looked at for removal

### ISSUE THREAD

**Issue by anoobindisguise** [CleverRaven/Cataclysm-DDA#64658]
### Is your feature request related to a problem? Please describe.

According to https://cataclysmdda.org/technology.html, "Cybernetics were brought to earth after the cataclysm, by the Exodii. A readily available CBM should not contain technology that can’t be explained by modern technology, except for the details of it being attached to the body without harm." We have a lot of bionics that do not meet this criteria right now.

- Bionic Claws and Monoblade: These magically never get dull and are totally unbreakable. There are other problems with these (gore-stained sword retracting back into your arm, extendable through clothing) but this is the most fundamental one.

- Probability Travel: Quantum tunneling through a wall is just magic, and it's not clear why it needs to be a cbm.

- Cloaking System: Uses LEDs to make you invisible, by blending you into the environment? The basic function is already basically impossible as you'd need to cover your entire body in cloaking tech, so it wouldn't work if you were wearing anything, and invisibility would also probably make you blind.

- Chain Lightning: Power cost would be orders of magnitude beyond anything you could feasibly supply with your body.

- Repair Nanobots: We don't have these IRL, even though nanomachines are a classic sci fi trope it's not something we can feasibly describe as anything other than magic healing robots. This is iffy tho as it could be a "lost exodii tech" that they no longer understand the principles of

- Sonic Resonator / Shockwave Unit / Flashbang Unit: All of these suffer from the problem of you being the epicenter of their functionality, it's not clear how you can have a flashbang centered around you without flashbanging yourself, or a shockwave that can vaporize rock (but somehow not harm anything living) or generate a ridiculously powerful shockwave that sends everything other than you flying.

- Time Dilation: No way to feasibly make your body move 10-20x as fast with the mental processing to match

- Autonomous Surgical Scalpels: AI driven dissection program which is far more efficient than a single surgeon's scalpel. Lots of issues like the scalpels wearing out and the significant contamination risk from them retracting back into you. They're also unreasonably effective as melee weapons.

- Thermal Dissipation: Putting heatsinks in your skin won't let you wade through lava and blazing infernos without damage

- Artificial Night Generator: It somehow generates exact opposite-amplitude lightwaves sufficient to neutralize all incoming light. Needless to say this is ludicrous even if it wasn't surgically implanted into your body (and covered up by clothing/skin/etc) and it should also make it impossible to see if no light can reach you but instead it just makes a sphere of "dark" around you and you can somehow see perfectly lit locations outside of it.

There are probably also others beyond these ones but these are the worst offenders.

### Solution you would like.

I don't know what to do about this. Some of them could be moved to Nether but even the internal storage CBM made more sense than some of these do. There are a *lot* of bionics that are not explainable with modern technology.

### Describe alternatives you have considered.

Change the standards for inclusion of bionics?

**Comment by Zireael07** [comment]
Agree with all of those EXCEPT Cloaking System (see https://interestingengineering.com/innovation/invisibility-cloaks-are-no-longer-just-science-fiction yes it would probably need tuned down a bit but I think it can stay)

**Comment by anoobindisguise** [comment]
> Agree with all of those EXCEPT Cloaking System (see https://interestingengineering.com/innovation/invisibility-cloaks-are-no-longer-just-science-fiction yes it would probably need tuned down a bit but I think it can stay)

So the thing about that is, that means the invisibility cloak *item* is probably OK, but not the actual bionic. Because it would need to cover your whole body including eyes, hair, etc, and you'd have to be naked.

**Comment by Venera3** [comment]
The blade is not much weirder than our normal weapons not dulling, and they could move over to limbs if we get to limbing bionics. Agree on the rest.

**Comment by Zireael07** [comment]
I thought the CBM was just "unspooling" the cloak from some compartment thingy hidden in/on your body, like at least one IRL cyborg has a compartment in his limb. If it's not, then yep it'd have to go

**Comment by GuardianDll** [comment]
I should point the whole text from the doc:

"Cybernetics were brought to earth after the cataclysm, by the Exodii. A **readily available CBM** should not contain technology that can’t be explained by modern technology, except for the details of it being attached to the body without harm... **Unique advanced CBMs may possess greater capabilities**, but access to these should be strictly gated through story advancement through one of several factions."

We don't really need to obsolete them, though some sort of reflavoring and locking only for VIP may be useful here

**Comment by andrewhr** [comment]
About potential solution, that's something I've commented on this thread https://github.com/CleverRaven/Cataclysm-DDA/issues/28273#issuecomment-1481582555.

TL;DR: rework CBM concept and implementation to have full cybernetic limbs with "mods and pockets" for extra functionality. This by itself will "fix" some of cited CBMs - blades and tools, for example, are just salvaged and adapted tools to be embedded in a cyber-arm. I argue that is also more aligned with Exodii concept of machinefication.

I agree with @anoobindisguise some other CBMs, like the cloak, could be moved to external items like some special single-purpose suit.

**Comment by sonphantrung** [comment]
> About potential solution, that's something I've commented on this thread [#28273 (comment)](https://github.com/CleverRaven/Cataclysm-DDA/issues/28273#issuecomment-1481582555).
> 
> TL;DR: rework CBM concept and implementation to have full cybernetic limbs with "mods and pockets" for extra functionality. This by itself will "fix" some of cited CBMs - blades and tools, for example, are just salvaged and adapted tools to be embedded in a cyber-arm. I argue that is also more aligned with Exodii concept of machinefication.
> 
> I agree with @anoobindisguise some other CBMs, like the cloak, could be moved to external items like some special single-purpose suit.
Well, the cloak [already exists!](https://nornagon.github.io/cdda-guide/#/item/optical_cloak)

**Comment by Daved27hundred** [comment]
With the retconning of CBMs to the exodii this was a long time coming. I have no objections to most of these, but if people insist on keeping some of the more questionable ones, consider making them modules of [the exodii's own biped framework](https://nornagon.github.io/cdda-guide/#/mutation/EXODII_BODY_1) which require the frame be on your body before it can be installed.

**Comment by Terrorforge** [comment]
> it's not clear why it needs to be a cbm.

This problem is kind of inherent to the concept of cybernetics in general. With most technology, there just isn't a whole lot of reason to stick it inside your body. Some technology (e.g. the surgical scalpels) could perhaps benefit from direct mental control, but in that case the reasonable thing to do is to install a minimally invasive neural interface that could potentially connect to any number of things. There is absolutely no reason to stick knives and rebreathers under your skin, except maybe for highly specialized applications like assassination and deep-sea diving.

In fact, apart from things with obvious medical utility, I find that magical nonsense like probability travel is by far the easiest to justify why it has to be a CBM, because you can just sort of technobabble at it. Probability travel can only affect matter within a very limited field, and therefore has to be spread throughout your tissues. Bam! Done.

**Comment by Mister-Cultist** [comment]
If people want cbms gone because of muh realism how about you make a branch mod that just removes them instead of getting rid of the fun for everyone?

**Comment by Raikiri** [comment]
Counter argument is if the game lore contradicts an interesting element of gameplay, then the lore needs to be changed, not the game.

That being said, some CBM's are just unreasonably overpowered for how little investment they need.

**Comment by Maleclypse** [comment]
"readily available"  They need to be moved to the weird nethereum cbm dungeon that's been on the roadmap.  Because that's where the roadmap says the weird ones you listed are going.
